recipes/orc/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, copy, get, rmdir, replace_in_file, mkdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.60.0 <2.0 || >=2.0.5"
+
+class OrcRecipe(ConanFile):
+    name = "orc"
+    description = "ORC is a self-describing type-aware columnar file format designed for Hadoop workloads"
+    license = "Apache-2.0"
+    homepage = "https://orc.apache.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("orc", "columnar", "file-format", "hadoop")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_tools": [True, False],
+        "build_avx512": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_tools": False,
+        "build_avx512": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+        }
+
+    @property
+    def _is_legacy_one_profile(self):
+        return not hasattr(self, "settings_build")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "apple-clang":
+            # AVX support is not enabled by default, might need to add -mavx512f to CXXFLAGS
+            del self.options.build_avx512
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("lz4/1.9.4")
+        self.requires("snappy/1.1.9")
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("zstd/1.5.5")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src", build_folder="build")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def build_requirements(self):
+        if not self._is_legacy_one_profile:
+            self.tool_requires("protobuf/<host_version>")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
```

Comment:
Totally a nitpick, but I recommend following the order used in the [CMake package template](https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py) for the recipe methods to keep the order of their appliation clearer for people new to Conan.